govlearn-api/src/main/java/com/unimuenster/govlearnapi/group/controller/GroupContentController.java

```diff
@@ -0,0 +1,119 @@
+import com.unimuenster.govlearnapi.user.service.AuthenticationService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.security.SecurityRequirement;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/api/v1/groups/content")
+@Slf4j
+public class GroupContentController {
+
+    private final AuthenticationService authenticationService;
+    private final GroupService groupService;
+
+    @Operation(
+            security = { @SecurityRequirement(name = "Authorization") },
+            description = "Add content for a group member, returns forbidden, if the current user is not admin of the group."
+    )
+    @PreAuthorize("hasAuthority('user')")
+    @PutMapping()
+    public ResponseEntity addContent(
+            @RequestBody AddContentToMemberWsTo addContentToMemberWsTo
+    ) {
+
+        UserEntity currentUser = authenticationService.getCurrentUser();
+
+        // Using then actual memeber Id here
+        Group group = groupService.findGroupByMemberId(addContentToMemberWsTo.getMemberId());
+
+        if (group == null) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
+        }
+
+        boolean userAdmin = groupService.isUserAdmin(currentUser, group.getId());
+
+        if (!userAdmin) {
+            return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
+        }
+
+        groupService.addContentToMember(addContentToMemberWsTo.getMemberId(), addContentToMemberWsTo.getCourseId());
+
+        return ResponseEntity.ok(Response.of(true));
+
+    }
+
+    @Operation(
+            security = { @SecurityRequirement(name = "Authorization") },
+            description = "Add content to an entire group, returns forbidden, if the current user is not admin of the group."
+    )
+    @PreAuthorize("hasAuthority('user')")
+    @PutMapping("/all")
+    public ResponseEntity addContentToAll(
+            @RequestBody AddContentToGroupWsTo addContentToGroupWsTo
+    ) {
+
+        UserEntity currentUser = authenticationService.getCurrentUser();
+
+        boolean userAdmin = groupService.isUserAdmin(currentUser, addContentToGroupWsTo.getGroupId());
+
+        if (!userAdmin) {
+            return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
+        }
+
+        groupService.addContentToGroup(addContentToGroupWsTo.getGroupId(), addContentToGroupWsTo.getCourseId());
+
+        return ResponseEntity.ok(Response.of(true));
+
+    }
+
+    @Operation(
+            security = { @SecurityRequirement(name = "Authorization") },
+            description = "Get content based on user."
+    )
+    @PreAuthorize("hasAuthority('user')")
+    @GetMapping("/{groupId}")
```

Comment:
Wirft internal server error, wenn user nicht in der Gruppe ist:
java.lang.RuntimeException: User is not in group